.github/workflows/ci.yml

```diff
@@ -0,0 +1,166 @@
+name: Continuous Integration
+on:
+  pull_request:
+  merge_group:
+    types: [checks_requested]
+  workflow_dispatch:
+
+jobs:
+  elixir:
+    uses: ./.github/workflows/elixir.yml
+  rust:
+    uses: ./.github/workflows/rust.yml
+  kotlin:
+    uses: ./.github/workflows/kotlin.yml
+  swift:
+    uses: ./.github/workflows/swift.yml
+  static-analysis:
+    uses: ./.github/workflows/static-analysis.yml
+  terraform:
+    uses: ./.github/workflows/terraform.yml
+    secrets: inherit
+
+  # We could build these in GCP with Cloud Build, but for now it's
+  # less overhead to keep things in GH actions. See work on building these
+  # in GCP with Cloud Build: https://github.com/firezone/firezone/pull/2234
+  build-images:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        include:
+          - image_name: api
+            cache_scope: elixir
+            build-args: |
+              APPLICATION_NAME=api
+          - image_name: web
+            cache_scope: elixir
+            build-args: |
+              APPLICATION_NAME=web
+          - image_name: gateway
+            cache_scope: rust
+            build-args: |
+              PACKAGE=gateway
+          - image_name: relay
+            cache_scope: rust
+            build-args: |
+              PACKAGE=relay
+    needs:
+      - rust
+      - elixir
+      - static-analysis
+    permissions:
+      contents: read
+      id-token: write
+    env:
+      # mark:automatic-version
+      VERSION: "1.20231001.0"
```

Comment:
Would it be better to read a `VERSION` file in the project root instead? Feels a bit safer than sting-replacing in entire project all the time.

Reply:
Yeah that was the thought I initially had, but Cargo files for example would still have to use this convention, and then we'd have two mechanisms to update the version. I thought since comments are universal, just sticking to one might keep things simpler.

I think the seds should be fairly stable -- they only match on mark:automatic-version and replace the next line's semver regex with ours.

The one edge case this will mess up is if we have a semantic version number in the line after mark:automatic-version that we don't want to replace.

Reply:
I think it should be fine to have two mechanisms if, the file will be more stable and produce fewer file diffs. And seeing both of them will be trivial - they will be part of the same makefile. At very least we can combine two approaches:

Add `elixir/VERSION`:
```
# mark:automatic-version
%{"version" => "1.20231001.0"}
```
and use it in Mix projects. However I don't see any benefits to keep the grep here if just writing value to a file is so much simpler.